### PR TITLE
topology-aware: correctly reconfigure implicit affinities for configuration changes.

### PR DESCRIPTION
**Pull request by klihub**
Fix the handling of implicit affinities corresponding to the `colocate{Pods,Namespaces}` configuration settings. Unregister and reregister disabled and enabled implicit affinities as necessary when we receive a new configuration.

### CODE REVIEW

File: pkg/resmgr/cache/cache.go [hunk 1]
```diff
@@ -390,6 +390,8 @@ type Cache interface {
 	EvaluateAffinity(*Affinity) map[string]int32
 	// AddImplicitAffinities adds a set of implicit affinities (added to all containers).
 	AddImplicitAffinities(map[string]ImplicitAffinity) error
+	// AddImplicitAffinities deletes a set of implicit affinities by name.
+	DeleteImplicitAffinities(names []string)
```
Comment (fmuyassarov) [inline pkg/resmgr/cache/cache.go:394]:
```suggestion
	// 	DeleteImplicitAffinities(names []string) deletes a set of implicit affinities by name.
	DeleteImplicitAffinities(names []string)
```

Comment (klihub) [reply pkg/resmgr/cache/cache.go]: Ouch, thanks for spotting that ! Fixed.

File: cmd/plugins/topology-aware/policy/topology-aware-policy.go [hunk 2]
```diff
@@ -484,6 +484,8 @@ func (p *policy) Reconfigure(newCfg interface{}) error {
 		return policyError("failed to reconfigure: %v", err)
 	}
 
+	p.registerImplicitAffinities()
```
Comment (fmuyassarov) [inline cmd/plugins/topology-aware/policy/topology-aware-policy.go:487]: since we are here, can we perhaps handle the error instead of ignoring here?

Comment (klihub) [reply cmd/plugins/topology-aware/policy/topology-aware-policy.go]: Okay. I added an error check to both call sites, `Setup` and `Reconfigure`.

**Review by askervin — APPROVED** [review]
LGTM

**Review by fmuyassarov — APPROVED** [review]

**Pull request merged**